### DISCUSSION TITLE
Event replay not working when the event path is changed

### DISCUSSION THREAD

**Discussion by planetarquasar** [spatie/laravel-event-sourcing#466] [Q&A]
Hello, can anyone help me, why the event replay failed if I change the event path? So recently the business logic demanded to remove one event, but instead of delete the event i want to mark as obsolate, so in the events folder i mad a new folder called deprecated and i moved in the event, i changed the namespace and updated the event_class_map in the config/event-sourcing 
from: 
```
'event_class_map' => [
'resource-number-updated' => Resource\Events\ResourceNumberUpdated::class]
```
to:
```
'event_class_map' => [
'resource-number-updated' => Resource\Events\Deprecated\ResourceNumberUpdated::class]
```
After the change i wanted to run the event replay and it returned this error:
`App\Support\JsonSerializer::deserialize(): Return value must be of type Spatie\EventSourcing\StoredEvents\ShouldBeStored, __PHP_Incomplete_Class returned`
So after the change it somehow returning the `__PHP_Incomplete_Class` instead of the actually class, what I defined in the config.

**Comment by harrysbaraini** [comment]
Event is still probably saved as `Resource\Events\ResourceNumberUpdated:` in the database. I guess that you should update those events to be 'resource-number-updated'.

**Reply by Gummibeer** [reply]
✅ That's the solution - same as morphs it persist the FQCN by default. So update that to your new value and done.
